cmd/ko/publish.go

```diff
@@ -93,7 +94,7 @@ func publishImages(importpaths []string, no *NameOptions, lo *LocalOptions) {
 				log.Fatalf("error setting up default image publisher: %v", err)
 			}
 		}
-		if _, err := pub.Publish(img, importpath); err != nil {
+		if _, err := pub.Publish(img, importpath, ta.Tags); err != nil {
```

Comment:
I think we should pass this to the constructor instead of the `Publish` call(s).

Reviewing your other PR, my secret hope was that we could eventually get this down to one constructor with different functional options, but I need to play around sometime...